### PR TITLE
Enhancement/Modal header title alignment

### LINKED ISSUE #1405: [Enhancement] Modal header title alignment
**Please add a short description of your enhancement request**
Align Modal header title text with progress cicle and "Close" icon.

**Describe the solution you'd like**
Align Modal header title text with progress cicle and "Close" icon.

**Describe alternatives you've considered**
Title is misaligned - bad visual experience :-)

**Additional context**
Modal-Header-Title-Issue
Modal-Header-Title-Issue-Desktop

### Kick Off:

- [x] Ensure the enhancement is `UX refined` and aligned with UX 
_The component and/or enhancement should be published and available in the [Kirby Styleguide on Zeplin](https://zpl.io/258pXGj)_
- [x] Ensure the enhancement has been `Tech refined` with @kirbydesign/kirby-guild and this issue is updated with a clear implementation description  
_This issue should be in the [Ready to do](https://github.com/kirbydesign/designsystem/projects/1#column-4590936) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1) before starting implementation)_
- [x] Assign yourself to this issue and move it to the [In progress](https://github.com/kirbydesign/designsystem/projects/1#column-4590937) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1)

### Code:

- [x] Create Feature Branch from [master branch](https://github.com/kirbydesign/designsystem/tree/master)
- [x] Create a draft implementation and push to Github
- [x] Ask a member of @kirbydesign/kirby-guild for a WIP review by creating a draft Pull Request
- [x] Implement unit tests
- [x] Update Cookbook Examples and Showcase, i.e. see [Radio](https://cookbook.kirby.design/home/showcase/radio)  
_Also remember to add any relevant new API documentation_

### Review:

- UX review:
  - [x] Ensure implementation is correct in relation to the UX design and the [Kirby Styleguide on Zeplin](https://zpl.io/258pXGj)
  - [ ] With UX agree on the version of the implementation
- Code review:
  - [ ] Open a pull request (or mark the existing draft PR as `Ready for review`) and ask @kirbydesign/kirby-guild for a review  
_Remember to add `closes #issueno` to the description of the PR._
  - [ ] Once approved, merge feature branch/PR to master
- [ ] Ask a member of @kirbydesign/kirby-guild to add a link to component showcase from Kirby Component Status and update the version number

:tada: Celebrate

### PR DESCRIPTION
**Pull request by MartinSkovsen**
This PR closes #1405

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Enhancement (to existing content)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## Which issue documents the current behaviour?
Currently, our modal header uses ion-title with full height.

Issue Number: #1400

## What is the new behavior?
A specific top spacing (padding) of 30px for desktop and 38px for phone.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by flemcito — APPROVED** [review]
Alignment seems fine now 👍 
Would be nice to solve it with flex alignment, but I see that the ion-elements make that impossible. So LGTM 
giphub

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.scss [hunk 1]
```diff
@@ -73,6 +73,16 @@ ion-header ion-toolbar {
   --border-width: 0;
   --background: transparent;
   --color: #{get-color('black')};
+
+  ion-title {
+    $spacingToTop: 30px;
```
Comment (flemcito) [inline libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.scss:78]: would be nice to use a defined size here - but I tried it out and came to the same conclusion as you did here

Comment (MartinSkovsen) [reply libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.scss:78]: I wanted to use the sizes, but after some trying, I figured, as you did, a locally scoped variable would be best. This speaks into a greater issue with the modal header (and how it is connected to page), as I've discussed a few times with @jakobe. I'm afraid my solution won't work well with supporting #1171, so we should decide whether to go with my solution and accept that the solution for #1400 has to be conditional or address the bigger issue with the modal header.

Comment (MartinSkovsen) [reply libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.scss:78]: For the record, in relation to the modal header, I consider this to be atleast a strike 2 (strike 1 was #1217). I'm unaware of previous strikes, and can't find any related issues or PRs, which could be considered 1 more strike.

Comment (jakobe) [inline libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.scss:78]: If you need a magic number (I'm not saying you do in this case 😉) it's a good point moving it into a named var. SCCS vars are not scoped though, så consider declaring var at top of file instead...

**Review by jakobe — CHANGES_REQUESTED** [review]
Seems like this is actually a bug in the ion-toolbar comp? I get the same vertical alignment problem if I set different top/bottom paddings in their [modal example](https://ionicframework.com/docs/api/modal) using [the official css custom properties for ion-toolbar](https://ionicframework.com/docs/api/toolbar#css-custom-properties) (`--padding-bottom`/`--padding-top`)
Consider recreating this issue and bug report for Ionic...
One workaround could be to leave the top/bottom padding on toolbar in sync (i.e. 16px for both) and the applying the additional spacing through the `padding-top` of ion-toolbar (which we are already modifying because of safe-area...)

**Comment by flemcito** [comment]
> Seems like this is actually a bug in the ion-toolbar comp? I get the same vertical alignment problem if I set different top/bottom paddings in their [modal example](https://ionicframework.com/docs/api/modal) using [the official css custom properties for ion-toolbar](https://ionicframework.com/docs/api/toolbar#css-custom-properties) (`--padding-bottom`/`--padding-top`)
> Consider recreating this issue and bug report for Ionic...
> One workaround could be to leave the top/bottom padding on toolbar in sync (i.e. 16px for both) and then apply the additional spacing through the `padding-top` of ion-toolbar (which we are already modifying because of safe-area...)

Hard to tell if it's a bug or a feature 😁  But it's true that padding-top and bottom need to be in sync for the alignment to be right

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.scss (hunk 1 above)
Comment (MartinSkovsen) [reply libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.scss:78]: Got it. Thank you for specificity on how variables in SCSS scope. Personally, I'd prefer to keep it there, as it is the only place we'd use it, and I'd also argue, for the same reason, that addding it to the top of the file adds extra noice to a already pretty populated modal-wrapper.component.scss. However, I sympathize with the convention of adding scss variables to top of file.

**Comment by jakobe** [comment]
> > Seems like this is actually a bug in the ion-toolbar comp? I get the same vertical alignment problem if I set different top/bottom paddings in their [modal example](https://ionicframework.com/docs/api/modal) using [the official css custom properties for ion-toolbar](https://ionicframework.com/docs/api/toolbar#css-custom-properties) (`--padding-bottom`/`--padding-top`)
> > Consider recreating this issue and bug report for Ionic...
> > One workaround could be to leave the top/bottom padding on toolbar in sync (i.e. 16px for both) and then apply the additional spacing through the `padding-top` of ion-toolbar (which we are already modifying because of safe-area...)
> 
> Hard to tell if it's a bug or a feature 😁 But it's true that padding-top and bottom need to be in sync for the alignment to be right

I would say it's a bug, as it also messes with their own layout:
image

**Comment by MartinSkovsen** [comment]
> Seems like this is actually a bug in the ion-toolbar comp? I get the same vertical alignment problem if I set different top/bottom paddings in their [modal example](https://ionicframework.com/docs/api/modal) using [the official css custom properties for ion-toolbar](https://ionicframework.com/docs/api/toolbar#css-custom-properties) (`--padding-bottom`/`--padding-top`)
> Consider recreating this issue and bug report for Ionic...
> One workaround could be to leave the top/bottom padding on toolbar in sync (i.e. 16px for both) and then apply the additional spacing through the `padding-top` of ion-toolbar (which we are already modifying because of safe-area...)

[Like this](https://github.com/kirbydesign/designsystem/pull/1406/files#diff-fb483ce96e0dfb7058a861e02f9a224f7f30d73bb8223c742fbaddf898a0b2a7)?

**Comment by MartinSkovsen** [comment]
> Seems like this is actually a bug in the ion-toolbar comp? I get the same vertical alignment problem if I set different top/bottom paddings in their modal example using the official css custom properties for ion-toolbar (--padding-bottom/--padding-top)
> Consider recreating this issue and bug report for Ionic...

[Ionic bug created](https://github.com/ionic-team/ionic-framework/issues/23072). Feel free to comment, correct me or otherwise contribute

**Review by jakobe — APPROVED** [review]

**Pull request merged**